fvm/src/call_manager/default.rs

```diff
@@ -617,12 +616,18 @@ where
             Some(addr) => addr,
             None => match to.payload() {
                 Payload::BLS(_) | Payload::Secp256k1(_) => {
+                    if read_only {
```

Comment:
Checked after checking the payload type so we return "not found" before "read only".